src/nodeUtils.ts

```diff
@@ -0,0 +1,68 @@
+const crypto = require('crypto');
+import utils = require('./utils');
+
+/**
+ * Utility functions that rely on Node.js built-in libraries.
+ */
+const nodeUtils = {
```

Comment:
Let's call this something more descriptive. `PlatformFeatures`/`PlatformFunction`/... .

We should also separate the ones that depend on platform from other general purpose utils.

Check out how HttpClient and HttpClient tests are structured. We define an interface and two implementations. Then we define a set of tests and run the same set of tests for both implementations.